src/main/java/com/ncc/aif/ValidateAIF.java

```diff
@@ -0,0 +1,311 @@
+        final ImmutableList<File> filesToValidate;
+        if (fileList.isPresent()) {
+            filesToValidate = edu.isi.nlp.files.FileUtils.loadFileList(fileList.get());
+        } else if (params.isPresent("kbToValidate")) {
+            filesToValidate = ImmutableList.of(params.getExistingFile("kbToValidate"));
+        } else {
+            throw new RuntimeException("Either kbToValidate or kbsToValidate must be specified");
+        }
+
+        boolean allValid = true;
+        for (File fileToValidate : filesToValidate) {
+            logger.info("Validating " + fileToValidate);
+            final Model dataToBeValidated = loadModel(Files.asCharSource(fileToValidate, Charsets.UTF_8)
+                    .openBufferedStream());
+            if (!validator.validateKB(dataToBeValidated)) {
+                logger.info("Validation of " + fileToValidate + " failed.");
+                allValid = false;
+            }
+        }
+
+        if (!allValid) {
+            // failure code if anything fails to validate
+            System.exit(1);
+        }
+        else {
+            logger.info("All KBs were valid.");
+        }
+    }
+
+    /**
+     * Returns whether or not the KB and hypotheses are valid
+     */
+    public boolean validateTA3(Model dataToBeValidated) {
+        final Model unionModel = ModelFactory.createUnion(domainModel, dataToBeValidated);
+        return validateKB(dataToBeValidated, unionModel)
+                && validateAgainstShacl(unionModel, ta3ShaclModel);
+    }
+
+    /**
+     * Returns whether or not the KB is valid
+     */
+    public boolean validateKB(Model dataToBeValidated) {
+        return validateKB(dataToBeValidated, null);
+    }
+
+    /**
+     * Returns whether or not the KB is valid
+     *
+     * @param dataToBeValidated KB to be validated
+     * @param union             unified KB if not null
+     */
+    public boolean validateKB(Model dataToBeValidated, Model union) {
+        // we unify the given KB with the background and domain KBs before validation
+        // this is required so that constraints like "the object of a type must be an
+        // entity type" will know what types are in fact entity types
+        final Model unionModel = (union == null) ? ModelFactory.createUnion(domainModel, dataToBeValidated) : union;
+
+        // we short-circuit because earlier validation failures may make later
+        // validation attempts misleading nonsense
+        return /*ensureEveryNamedNodeHasARdfType(dataToBeValidated)
+                &&*/ validateAgainstShacl(unionModel, shaclModel)
+                && ensureConfidencesInZeroOne(unionModel)
+                && ensureEveryEntityAndEventHasAType(unionModel);
+    }
+
+    private static final String ENSURE_EVERY_NAMED_NODE_HAS_A_TYPE_SPARQL_QUERY = "\"\n" +
+            "    PREFIX rdf:\n" +
+            "\n" +
+            "    <$ {\n" +
```

Comment:
You need to replace ${} with the variables they represent (e.g. ${RDF.uri} becomes "<" + RDF.uri + ">\n")

Reply:
Oh goodness...this is what the IDE did initially when I copy/pasted the strings...I fixed the others like this but somehow missed this one.  Thanks!